POSYDON-code/POSYDON-public
# Add unit tests: grids

This PR will get unit tests for the `grids` module.

- [x] `downsampling.py`
- [x] `io.py`
- [x] `MODELS.py`
- [x] `post_processing.py`
- [x] `psygrid.py`
- [x] `scrubbing.py`
- [x] `termination_flags.py`
- [x] update `continuous_integration.yml`
- [x] tested on small pop-syn

Some improvements on the code:

- adding/editing docstrings
- correcting typos (e.g. wrong variable names, wrong strings/values, wrong error/warning types)
- add missing parts (e.g. allow zipped files, make type conversions, missing `raise`, avoid division by zero)
- avoid impossible branches (e.g. directly after you added a key to a dictionary, no need to check for the key)
- add SN data to unit test directory and point the `PATH_TO_POSYDON_DATA` in the tests there.
- apply change on missing and new objects for utils tests as well
- add Runtimeerror in case a system tool, e.g. gzip, fails (although in tests of utils)
